pages/faq-how-to-buy-fresh-flowers-and-gifts-in-lagos-and-abuja.tsx

```diff
@@ -147,8 +148,40 @@ const schemaProperties = {
 const Index: FunctionComponent<{ featuredFlowers: Product[] }> = ({
   featuredFlowers
 }) => {
-  const [activeContent, setActiveContent] = useState<ContentLink | null>(null);
+  // const [activeContent, setActiveContent] = useState<ContentLink | null>(null);
+  const [openQuestions, setOpenQuestions] = useState<OpenQuestions>({});
+  const questionList = schemaProperties.mainEntity;
+  interface OpenQuestions {
+    [key: number]: boolean;
+  }
+  const toggleAnswer = (index: number) => {
+    setOpenQuestions((prevOpenQuestions) => ({
+      ...prevOpenQuestions,
+      [index]: !prevOpenQuestions[index],
+    }));
+  };
+  
+  const questions = questionList.map((question, index) => {
+    const isAnswerOpen = openQuestions[index] || false;
 
+    return (
+      <div key={index}>
+        <div className={styles["faq-question"]} onClick={() => toggleAnswer(index)}>
+          <p className={`${styles["question-text"]} ${isAnswerOpen ? styles["question-text-open"] : ''}`}>{question.name}</p>
+          <div className={`${styles["icon-container"]} ${isAnswerOpen ? styles["icon-container-open"] : ''}`}>
+            <img
+              src={isAnswerOpen ? "/icons/minus_circle_outline.svg" : "/icons/plus_circle_outline.png"}
+              alt="expand-question-icon"
+              className={styles["expand-icon"]}
+            />
+          </div>
+        </div>
+        {isAnswerOpen && (
+          <div className={styles["faq-answer"]}>{question.acceptedAnswer.text}</div>
+        )}
```

Comment:
This div should have a different background color when the section is expanded. Check the design for the color. Also ensure that there's a set CSS transition for the background change

Reply:
done